### PR TITLE
workspace-module changes

### PR DESCRIPTION
**Pull request by labhvam5**

### CODE REVIEW

File: apps/fyle/tasks.py [hunk 1]
```diff
@@ -77,20 +77,25 @@ def async_create_expense_groups(workspace_id: int, fund_source: List[str], task_
...
+            )
+
+            if 'CCC' in fund_source:
+                expenses += platform.expenses.get(
+                source_account_type=['PERSONAL_CORPORATE_CREDIT_CARD_ACCOUNT'],
+                state=expense_group_settings.ccc_expense_state,
+                settled_at=last_synced_at if expense_group_settings.ccc_expense_state == 'PAYMENT_PROCESSING' else None,
                 filter_credit_expenses=filter_credit_expenses,
-                last_paid_at=last_synced_at if expense_group_settings.expense_state == 'PAID' else None
+                last_paid_at=last_synced_at if expense_group_settings.ccc_expense_state == 'PAID' else None
```
[the remark this answers is not in the document]
Comment (ashwin1111) [reply apps/fyle/tasks.py:98]: this is to maintain stuffs for old app as well as new app

File: apps/workspaces/views.py [hunk 2]
```diff
@@ -154,12 +170,14 @@ def post(self, request, **kwargs):
             fyle_user = get_fyle_admin(tokens['access_token'], None)
             org_name = fyle_user['data']['org']['name']
             org_id = fyle_user['data']['org']['id']
+            org_currency = fyle_user['data']['org']['currency']
```
Comment (ashwin1111) [inline apps/workspaces/views.py:173]: same

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: apps/workspaces/models.py [hunk 3]
```diff
@@ -38,6 +42,10 @@ class Workspace(models.Model):
     name = models.CharField(max_length=255, help_text='Name of the workspace')
     user = models.ManyToManyField(User, help_text='Reference to users table')
     fyle_org_id = models.CharField(max_length=255, help_text='org id', unique=True)
+    fyle_currency = models.CharField(max_length=5, help_text='Fyle Currency', null=True)
```
[the remark this answers is not in the document]
Comment (labhvam5) [reply apps/workspaces/models.py:45]: No we need to do it along with these changes only .

File: apps/workspaces/apis/export_settings/serializers.py [hunk 4]
```diff
@@ -132,6 +132,12 @@ def update(self, instance, validated):
         if 'expense_state' in expense_group_settings and not expense_group_settings['expense_state']:
             expense_group_settings['expense_state'] = 'PAYMENT_PROCESSING'
 
+        if not expense_group_settings['reimbursable_export_date_type']:
+            expense_group_settings['reimbursable_export_date_type'] = 'current_date'
+
+        if not expense_group_settings['ccc_export_date_type']:
+            expense_group_settings['ccc_export_date_type'] = 'current_date'
```
[the remark this answers is not in the document]
Comment (labhvam5) [reply apps/workspaces/apis/export_settings/serializers.py:139]: yes .

File: scripts/python/001_fill_xero_and_fyle_currency.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+workspaces = Workspace.objects.filter(Q(fyle_currency=None) | Q(xero_currency=None)).all()
+
+for workspace in workspaces:
+    expense = Expense.objects.filter(org_id=workspace.fyle_org_id).first()
+    if workspace.fyle_currency == None and expense:
+        workspace.fyle_currency = expense.currency
+    elif workspace.fyle_currency == None:
+        fyle_credentials = FyleCredential.objects.get(workspace_id=workspace.id)
+        platform = PlatformConnector(fyle_credentials)
```
Comment (ashwin1111) [inline scripts/python/001_fill_xero_and_fyle_currency.py:16]: Do this in a try catch block, expired tokens shouldn't affect the script

File: scripts/python/001_fill_xero_and_fyle_currency.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        my_profile = platform.connection.v1beta.spender.my_profile.get()
+        fyle_currency = my_profile['data']['org']['currency']
+        workspace.fyle_currency = fyle_currency
+
+    if workspace.xero_currency == None:
+        xero_credentials = XeroCredentials.objects.filter(workspace_id=workspace.id).first()
+        xero_connector = XeroConnector(xero_credentials, workspace_id=workspace.id)
```
Comment (ashwin1111) [inline scripts/python/001_fill_xero_and_fyle_currency.py:23]: Do this in a try catch block, expired tokens shouldn't affect the script

File: scripts/python/001_fill_xero_and_fyle_currency.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+from apps.workspaces.models import Workspace
+from apps.fyle.models import Expense
+from apps.xero.utils import XeroConnector
+from apps.workspaces.models import XeroCredentials, FyleCredential
+from fyle_integrations_platform_connector import PlatformConnector
+from django.db.models import Q
+
+workspaces = Workspace.objects.filter(Q(fyle_currency=None) | Q(xero_currency=None)).all()
```
Comment (ashwin1111) [inline scripts/python/001_fill_xero_and_fyle_currency.py:8]: Write an sql script to fill dummy currency - "USD" for all demo and test orgs. We can run that first and then run this script.

File: sql/scripts/011-fill-xero-and-fyle-currency-demo-orgs.sql [hunk 8]
```diff
@@ -0,0 +1,15 @@
+rollback;
+begin;
+
+update 
+  workspaces 
+set 
+  xero_currency = 'USD', 
+  fyle_currency = 'USD' 
+where 
+  id in (
```
Comment (ashwin1111) [inline sql/scripts/011-fill-xero-and-fyle-currency-demo-orgs.sql:10]: id not in :P

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**